javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -359,9 +373,18 @@ public boolean isEmpty() {
             this.value = value;
         }
 
+        private boolean itsMe(int keyHashCode, K key) {
```

Comment:
Two minor points:

* Let's name it `equals`, it is an optimized form of `equals(Object)` that first tests the `hash`.
* Let's also call the first param `keyHash` or rename the instance variable to `hashCode` to have consistent naming.